core/trino-main/src/main/java/io/trino/operator/TaskContext.java

```diff
@@ -590,4 +601,33 @@ public QueryContext getQueryContext()
     {
         return queryContext;
     }
+
+    public void addDynamicFilter(Map<DynamicFilterId, Domain> dynamicFilterDomains)
+    {
+        if (dynamicFilterDomains.isEmpty()) {
+            return;
+        }
+        synchronized (coordinatorDynamicFilters) {
+            ImmutableMap.Builder<DynamicFilterId, Domain> domainsForConsumer = ImmutableMap.builder();
+            dynamicFilterDomains.forEach((dynamicFilterId, dynamicFilterDomain) -> {
+                Domain previousDomain = coordinatorDynamicFilters.put(dynamicFilterId, dynamicFilterDomain);
+                // Coordinator may re-send dynamicFilterDomain if sendUpdate request fails
+                // It's possible that the request failed after the DF was already collected here
+                if (previousDomain != null && previousDomain.equals(dynamicFilterDomain)) {
+                    return;
+                }
+                verify(previousDomain == null, "Dynamic filter %s already collected", dynamicFilterId);
+                domainsForConsumer.put(dynamicFilterId, dynamicFilterDomain);
+            });
+            coordinatorDynamicFilterConsumers.forEach(consumer -> consumer.accept(domainsForConsumer.build()));
+        }
+    }
+
+    public void addCoordinatorDynamicFiltersConsumer(Consumer<Map<DynamicFilterId, Domain>> consumer)
```

Comment:
Move `io.trino.sql.planner.LocalExecutionPlanner.LocalExecutionPlanContext#dynamicFiltersCollector` to `TaskContext` instead of extra layer of indirection